### DISCUSSION TITLE
Support for Deezer as Music provider

### DISCUSSION THREAD

**Discussion by marcelveldt** [music-assistant/support#434] [Feature requests and ideas]
Add Deezer support to music providers

## DEEZER IS AVAILABLE https://music-assistant.io/music-providers/deezer/

**Comment by Clooos** [comment]
I can't agree more 🙏

**Comment by jonbng** [comment]
I am willing to spend some time on this

**Reply by marcelveldt** [reply]
That is great to hear! Be aware that I'm, working on a partial refactor after which it should be easier to adopt new music providers. This refactor should be ready in a week or two. Please contact me on discord if you are willing to code a music provider for Deezer.

**Reply by labaland** [reply]
Please man how is it going? 
It is evern possible to download 320kbps / FLAC from deezer with 3thirdparty programs. Must be easy to add support then. But i dont know :) 
Thank you!

**Reply by jonbng** [reply]
Hi! Its going good-ish. I understand its annoying to wait so long, but I am still working on it. I hope to finish playback (and maybe OAUTH) this weekend, but i cant promise anything. If you have experience in coding python you are welcome to help, you can reach me in the Discord server. :slightly_smiling_face: 
> It is evern possible to download 320kbps / FLAC from deezer with 3thirdparty programs. Must be easy to add support then.

Due to TOS we aren't allowed to download them, we have to decipher them JIT. And streaming the songs isn't everything needed for it to work. 
Currently we plan to use this: https://github.com/yne/dzr as inspiration for streaming but if there is something better we could switch..

**Reply by marcelveldt** [reply]
@labaland if you post again with links to illegal download scripts I will ban you. Also it is not fair to make statements like "must be easy to support them". That someone hacks together an illegal download script is not the same as implementing a full fledged music provider in MA that respects the TOS of a music provider. I have edited your posts because I do not want anything to do with scripts/sites that are involved with illegal downloading. Respect the artists please

**Comment by labaland** [comment]
Please add deezer as provider!

It is evern possible to download 320kbps / FLAC from deezer with 3thirdparty programs. Must be easy to add support then?? i dont know :)

Maybe theese sites/files can help some?

I pray for it 🙏

@OzGav . Im Sorry for new thread.

**Reply by marcelveldt** [reply]
@labaland I do not want links to thart sort of downloaders here, I have edited your post

**Comment by jonbng** [comment]
As of version 2.0.0 beta 30 Music Assistant has support for Deezer as a music provider. More information here: https://github.com/orgs/music-assistant/discussions/1245

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
